edx/edx-app-ios
# iOS - Install Segment to firebase integration

### Description

[LEARNER-7256](https://openedx.atlassian.net/browse/LEARNER-7256)

This PR has segment to firebase integration as per [Segment document](https://segment.com/docs/destinations/firebase/). The pod 'Segment-Firebase' requires Analytics minimum version 3.2 so Analytics has updated from 3.0.0 - 3.6.10 the latest version, Segment-GoogleAnalytics pod had dependency on Analytics so it also get updated from 1.0.0 - 1.2.0.  On Segment console the firebase destination has added on source "mobile-prod-edx-ios
IOS Mobile".

### Notes

- Firebase Analytics will work as standalone. 
- If segment and firebase analytics both are enable the events will be sent to segment/googleAnalytics/Firebase consoles through segment library.
- If segment is disable and firebase analytics is enable then firebase analytics event will be sent on firebase console as through firebase library.


### How to test this PR
Events should appear on segment or firebase console.
Example:
Open Course dashboard screen and go to course dashboard tab. The even should appear on segment console as well as firebase console.
